### PR TITLE
link entities to invoices

### PR DESCRIPTION
**Pull request by Conni2461**
## Description

Continuation of #2414 (did rebase it) based on this discussion: https://github.com/kevinpapst/kimai2/discussions/2630

Not sure if thats the intended way

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [x] I verified that my code applies to the guidelines (`composer code-check`)
- [ ] I updated the documentation (see [here](https://github.com/kimai/www.kimai.org/tree/master/_documentation))
- [x] I agree that this code is used in Kimai and will be published under the [MIT license](https://github.com/kevinpapst/kimai2/blob/master/LICENSE)

### CODE REVIEW

File: src/Repository/TimesheetInvoiceItemRepository.php [hunk 1]
```diff
@@ -35,6 +36,27 @@ public function getInvoiceItemsForQuery(InvoiceQuery $query): iterable
         return $this->repository->getTimesheetsForQuery($query, true);
     }
 
+    /**
+     * @param Invoice $invoice
+     * @return InvoiceItemInterface[]
+     */
+    public function getInvoiceItemsByInvoice(Invoice $invoice): iterable
+    {
+        return this->repository->getTimesheetsByInvoice(invoice, true);
```
Comment (kevinpapst) [inline src/Repository/TimesheetInvoiceItemRepository.php:45]: Did you test that code?

**Review by kevinpapst — CHANGES_REQUESTED** [review]

File: src/Repository/TimesheetInvoiceItemRepository.php (hunk 1 above)
Comment (Conni2461) [reply src/Repository/TimesheetInvoiceItemRepository.php:45]: Not yet. I planned to do that later today, if i find the time. Thanks for looking over it

Comment (kevinpapst) [reply src/Repository/TimesheetInvoiceItemRepository.php:45]: Okay, as I receive an email on every action (commit ...) can you send me a ping once you are ready? 
Just so I know that I can start reviewing instead.

**Comment by kevinpapst** [comment]
I am working on a huge refactoring of this one, including linking timesheets to multiple invoices (think canceled invoices) and a generation of a "Timesheet PDF" to provide to the customer.
So please be patient, you'll get a notification once I am done with it.

**Comment by kevinpapst** [comment]
As written before, this PR is replaced by #3080 
I already closed #2995 
If you need this feature, please test #3080 and add changes if you need them. There is no `receiveByInvoice()` method right now as I don't have a use-case.
Other related changes were merged in #3079 

Big thanks for spending the time to improve Kimai. Sorry that this PR did not make it, because I changed my mind during the work and sorry for the too long waiting period.

**Pull request closed without merging**